src/NumSharp.Core/Backends/Default/Math/Default.Round.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading.Tasks;
+using DecimalMath;
+using NumSharp.Utilities;
+
+namespace NumSharp.Backends
+{
+    public partial class DefaultEngine
+    {
+        public override NDArray Round(in NDArray nd, Type dtype) => Round(nd, dtype?.GetTypeCode());
+
+        public override NDArray Round(in NDArray nd, NPTypeCode? typeCode = null)
+        {
+            if (nd.size == 0)
+                return nd.Clone();
+
+            var @out = Cast(nd, ResolveUnaryReturnType(nd, typeCode), copy: true);
```

Comment:
Round does not resolve dtype like this, it supports all dtypes and return is equivalent to input (to verify read docs)
Proof:
```python
import numpy as np

x = np.array([1, 2, 3])
print(x.dtype)
x = np.round(x)
print(x.dtype)
print(x.shape)
print(x)
```